test-harness/tests/conftest.py

```diff
@@ -246,9 +246,20 @@ def f(uid, *args, **kwargs):
 
 
 @pytest.fixture(scope='session')
+def superuser_user_header(valid_jwt_generator):
+    """This fixture further simplifies JWT handling by providing a ready-to-use
+    headers with a valid  superuser JSON Web Token for `requests` module to
+    use"""
+    token = valid_jwt_generator(uid='root')
```

Comment:
This has a couple of layers. I will try to answer your questions in one go:

> I'm curious: how does auth get out of the picture with the superuser header?

This is hard to see in Open AR test harness, but EE has a Bouncer Mock that supports giving/revoking permissions (https://github.com/mesosphere/adminrouter/blob/prozlach/synchronize_with_open-tests_for_ee/test-harness/modules/mocker/endpoints/ee/iam.py). `root/superuser/admin/etc...` does not matter much for Open as all users are equal, for EE root is:
* always there, by default
* it does not care about permissions that endpoint require

Thus you can say "get out of the picture" because everything is permitted for "root", both in EE and Open.

The reason why there is no differentiation between Open and EE when it comes to user names is that some tests are used by both EE and Open (`overlay` concept) and it does not seem be woth it to use different user - Open does not care how user is named as long it's there, EE needs this user to be 'root' in order to "get out of the picture".

As for the reason why we want to `get out of the picture` at all - auth tests are done differently for EE and Open and should be tested independently anyway. So we split testing if endpoint is properly authn/authz-ing from testing if endpoint does what it is supposed to.

> Where does the relation between superuser and root come from?

This could be anything really: `admin`, `superuser`, `bootstrapuser`, `root`. The `root` came to my mind as a best choice because people writting tests are accustomed to the fact that `root` can do anything. It would be hard to explain `bootstrapuser` to outsiders, `admin` is not as strong as `root` I think.
If you insist and have good arguments we can change it, otherwise I would like to leave it as-is.

> That concept is not known in DC/OS.

Think `user which belongs to "superusers" group`.

> Is that a relation built into the test harness/mock system?

It's a relation that the mock/test-harness tries to mimick.

I hope it clarifies a bit.